tool/tctl/common/tctl.go

```diff
@@ -257,6 +258,19 @@ func TryRun(commands []CLICommand, args []string) error {
 			PromptConfig: *promptCfg,
 		})
 	})
+	client.SetSSOMFACeremonyConstructor(func(ctx context.Context) (mfa.SSOMFACeremony, error) {
+		rdConfig := sso.RedirectorConfig{
+			ProxyAddr: proxyAddr,
+		}
+
+		rd, err := sso.NewRedirector(rdConfig)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		context.AfterFunc(ctx, rd.Close)
```

Comment:
Same comment as the other AfterFunc - it's a bit surprising that the ctx would have an effect after the function call ended.